integration/e2e_test.go

```diff
@@ -299,3 +302,59 @@ func deployMicrobot(t testing.TB) {
 		}
 	}
 }
+
+func upgradeCluster(t testing.TB, version string) {
+	t.Logf("Upgrading cluster to %s", version)
+	// Patch contol plane machine upgrades.
+	patch_command := fmt.Sprintf("kubectl patch --type=merge microk8scontrolplanes.controlplane.cluster.x-k8s.io microk8s-openstack-control-plane --patch {\"spec\":{\"version\":\"%s\"}}", version)
+	command := strings.Fields(patch_command)
+	cmd := exec.Command(command[0], command[1:]...)
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		t.Error(string(output))
+		t.Fatalf("Failed to merge the patch to control plane. %s", err)
+	}
+
+	// Patch worker machine upgrades.
+	patch_command = fmt.Sprintf("kubectl patch --type=merge machinedeployments.cluster.x-k8s.io microk8s-openstack-md-0 --patch {\"spec\":{\"template\":{\"spec\":{\"version\":\"%s\"}}}}", version)
```

Comment:
Same here, this is openstack specific.

Reply:
Please define this as a `[]string{arg1, arg2, ...}`. No need to split strings, and it makes the json argument difficult to parse. For the JSON argument, also consider raw strings, e.g.

```
fmt.Sprintf(`{"spec":{"template":{"spec":{"version":"%s"}}}}`, version)
```

Reply:
Is using environment variables for this a better alternative than parsing the manifest file to extract these names? I think for such a trivial task parsing is a bit overkill. WDYT?

Reply:
@neoaggelos the raw strings did pose a problem, but in this current form, they work as expected. Is there any additional benefit apart from good practice to write JSON arguments separately?

Reply:
you can make this an environment variable I guess. The test "runner" controls the whole cluster manifest, specifying the machine deployment to scale/upgrade/_act_ on sounds fine to me.

> @neoaggelos the raw strings did pose a problem, but in this current form, they work as expected. Is there any additional benefit apart from good practice to write JSON arguments separately?

What kind of problem did you encounter?

In general, I'm heavily against `strings.Fields("...")` because if anyone at any point adds spaces to the arguments then everything falls apart in unexpected ways.